int32 shootPower
int32 shootAngle
